src/test/transaction_tests.cpp

```diff
@@ -370,6 +369,40 @@ BOOST_AUTO_TEST_CASE(tx_invalid)
     }
 }
 
+BOOST_AUTO_TEST_CASE(tx_no_inputs)
+{
+    CMutableTransaction empty;
+
+    TxValidationState state;
+    BOOST_CHECK_MESSAGE(!CheckTransaction(CTransaction(empty), state), "Transaction with no inputs should be invalid.");
+    BOOST_CHECK(state.GetRejectReason() == "bad-txns-vin-empty");
+}
+
+BOOST_AUTO_TEST_CASE(tx_oversized)
+{
+    auto maxPayloadSize = MAX_BLOCK_WEIGHT / WITNESS_SCALE_FACTOR - 70;
+    {
+        CMutableTransaction tx;
+        tx.vin.resize(1);
+        tx.vout.emplace_back(1, CScript() << OP_RETURN << std::vector<unsigned char>(maxPayloadSize));
+
+        TxValidationState state;
+        CheckTransaction(CTransaction(tx), state);
+        BOOST_CHECK(state.GetRejectReason() != "bad-txns-oversize");
+    }
+
+    maxPayloadSize += 1;
+    {
+        CMutableTransaction tx;
+        tx.vin.resize(1);
+        tx.vout.emplace_back(1, CScript() << OP_RETURN << std::vector<unsigned char>(maxPayloadSize));
+
+        TxValidationState state;
+        BOOST_CHECK_MESSAGE(!CheckTransaction(CTransaction(tx), state), "Oversized transaction should be invalid");
+        BOOST_CHECK(state.GetRejectReason() == "bad-txns-oversize");
+    }
+}
```

Comment:
Nice to have: To reduce duplication by using lambda expressions and captures.

Reply:
We only have 3 matching lines, so extracting them would result in slightly more complicated code.
If the reviewers insist, I don't mind doing it, see https://github.com/bitcoin/bitcoin/pull/29862#discussion_r1643367751

Reply:
I've done this as well as part of the above change.